tests/math/test_expectation_value.py

```diff
@@ -0,0 +1,211 @@
+# Copyright 2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit tests for differentiable matrix-vector expectation values.
+"""
+
+import numpy as onp
+import pytest
+
+import pennylane as qml
+from pennylane import numpy as np
+
+pytestmark = pytest.mark.all_interfaces
+
+tf = pytest.importorskip("tensorflow", minversion="2.1")
+torch = pytest.importorskip("torch")
+jax = pytest.importorskip("jax")
+jnp = pytest.importorskip("jax.numpy")
+
+
+class TestExpectationValueMath:
+    """Tests for Expectation value of a operator for a state vector."""
+
+    ops_vs_vecstates = [
+        ([[1, 0], [0, 0]], [1, 0], 1),
+        ([[0, 1], [1, 0]], [0, 1], 0),
+        ([[0.5, 0.5], [0.5, 0.5]], [1, 1] / np.sqrt(2), 1),
+        (
+            [[0.40975111, 0.40751457], [0.40751457, 0.59024889]],
+            [0.8660254, 0.5],
+            0.8077935208042251,
+        ),
+        (
+            [[1, 0, 0, 0], [0, 1, 0, 0], [0, 0, -1, 0], [0, 0, 0, -1]],
+            [1, 0, 1, 0] / np.sqrt(2),
+            0,
+        ),
+    ]
+
+    array_funcs = [
+        lambda x: x,
+        onp.array,
+        np.array,
+        jnp.array,
+        torch.tensor,
+        tf.Variable,
+        tf.constant,
+    ]
+
+    check_state = [True, False]
+    check_operator = [True, False]
+
+    @pytest.mark.parametrize("operator_and_states", ops_vs_vecstates)
+    @pytest.mark.parametrize("check_state", check_state)
+    @pytest.mark.parametrize("check_operator", check_operator)
```

Comment:
Probably do not need to parametrize `check_state` and `check_operator`. This results in four repeated executions of the same test case. As long as they are tested somewhere (which I see that it is), it should be sufficient.